scib/metrics/pcr.py

```diff
@@ -216,18 +271,39 @@ def pc_regression(
     else:
         if verbose:
             print("one-hot encode categorical values")
-        covariate = pd.get_dummies(covariate)
+        covariate = pd.get_dummies(covariate).to_numpy()
 
     # fit linear model for n_comps PCs
-    r2 = []
-    for i in range(n_comps):
-        pc = X_pca[:, [i]]
-        lm = LinearRegression()
-        lm.fit(covariate, pc)
-        r2_score = np.maximum(0, lm.score(covariate, pc))
-        r2.append(r2_score)
+    if verbose:
+        print(f"Use {n_threads} threads for regression...")
+    if n_threads == 1:
+        r2 = []
+        for i in tqdm(range(n_comps), total=n_comps):
+            r2_score = linreg_method(X=covariate, y=X_pca[:, [i]])
+            r2.append(np.maximum(0, r2_score))
+    else:
+        with ThreadPoolExecutor(max_workers=n_threads) as executor:
+            run_r2 = executor.map(linreg_method, [covariate] * n_comps, X_pca.T)
+            r2 = list(tqdm(run_r2, total=n_comps))
 
     Var = pca_var / sum(pca_var) * 100
     R2Var = sum(r2 * Var) / 100
 
     return R2Var
+
+
+def linreg_sklearn(X, y):
+    from sklearn.linear_model import LinearRegression
+
+    lm = LinearRegression()
```

Comment:
There is an `n_jobs` parameter with a default 1. Setting it to the number of threads would allow faster calculation with sklearn only

Reply:
You mean for the sklearn function? I would avoid threading within an already threaded loop, since that would square the number of threads requested in the worst case